### ISSUE TITLE
mediaUriElement: RTSP streaming freezes after some seconds without firing any event

### ISSUE THREAD

**Issue by KodeKooker** [Sascha-L/WPF-MediaKit#46]
As written in the title, I've tried setting the mediaUriElement source to the RTSP h264 stream provided by a AXIS IP Camera; the stream starts for 1-2 seconds, then it freezes, without firing any event (mediafailed/mediaended/mediaclosed). I see that the mediaposition continues changing but the player is frozen on the last showed frame. If I set the source again, "mediaended" and then "mediaopened" are fired and the stream starts playing again for 1-2 seconds then freezes again.
I've already tried setting "Loop" property to true.
I've tried setting the source to another public RTSP h264 stream (es. bigbuckbunny), without having any problem. But with other WPF components (es. FFMpeg based) my AXIS IP camera RTSP stream shows correctly without freezing.
What could it be?
Thanks

**Comment by jluisestrada** [comment]
Maybe the graph. The MediaKit creates a basic graph using the automated linking when adding the filters, so, somethimes it doesnt load the proper filters.


Load the camera's software and execute it, then, open graphedit and load the runtime graphs. And after that compare it with the graph created by the media kit


________________________________

**Comment by KodeKooker** [comment]
I'm choosing which media component I will integrate in my application, so, sorry, I don't have time to debug it at such a low level... please notify me if this (supposed) bug is going to be looked into any time soon, thanks

**Comment by jluisestrada** [comment]
http://i.dailymail.co.uk/i/pix/2013/09/26/article-0-18472E1B00000578-928_634x354.jpg

[http://i.dailymail.co.uk/i/pix/2013/09/26/article-0-18472E1B00000578-928_634x354.jpg]


________________________________

**Comment by xmedeko** [comment]
@jluisestrada Your previous post contain some pictures, are they related to this issue or is it a mistake?

**Comment by xmedeko** [comment]
@KodeKooker it's hard to tell if it's a bug or some other problem with your installation. The [StreamPlayerControl and WebCameraControl](https://github.com/jacobbo/WebEye) look interesting, it's main differences are: 1) FFMpeg based, 2) uses HwndHost (airspace issues).
There are a few others WPF video projects - players. Unfortunately, each of them have some bugs, deficiencies. There is no clearly the best WPF video player.
@jacobbo what about to improve the WPF Media Kit instead of the starting own WebCameraControl?

**Comment by jacobbo** [comment]
Sorry, can't help with your issue as I'm not familiar with LAV filters.

**Comment by xmedeko** [comment]
@jacobbo yep, but would it be possible to improve the WPF-MediaKit to be able to use FFMpeg for decoding and it's `D3DRenderer` for rendering?

**Comment by xmedeko** [comment]
@KodeKooker try the [WebCameraControl](https://github.com/jacobbo/WebEye/tree/master/WebCameraControl), it's based on the DirectShow, too. If you have same problems, then the problem is in the DirectShow - the DirectShow filters probably. Try installing the newest K-Lite codecs, see Wiki https://github.com/Sascha-L/WPF-MediaKit/wiki/Codecs
